File: app/src/main/java/ua/hospes/rtm/core/di/components/AppComponent.java
package ua.hospes.rtm.core.di.components;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import ua.hospes.rtm.App;
import ua.hospes.rtm.core.di.module.ActivitiesModule;
import ua.hospes.rtm.core.di.module.AppModule;

@Singleton
@Component(modules = {AndroidInjectionModule.class, AppModule.class, ActivitiesModule.class})
public interface AppComponent {
    void inject(App app);
}

File: app/src/main/java/ua/hospes/rtm/core/di/AppComponent.kt
package ua.hospes.rtm.core.di

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import ua.hospes.rtm.App
import ua.hospes.rtm.core.di.module.ActivitiesModule
import ua.hospes.rtm.core.di.module.AppModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    AppModule::class,
    ActivitiesModule::class])
interface AppComponent : AndroidInjector<App> {
    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<App>()
}